### PR TITLE
Uses a jittered exponential backoff

### PR DESCRIPTION
**Pull request by ldez**
- during tests, allow to support more than 90% of bad nonce with a minimum of attempts.
   - before with the simple retry system (5 attempts): ~50%
- partial fix for #486 and #793

### CODE REVIEW

File: acme/api/api.go [hunk 1]
```diff
@@ -64,34 +67,46 @@ func (a *Core) post(uri string, reqBody, response interface{}) (*http.Response,
...
-			log.Infof("nonce error retry: %s", err)
-			resp, err = a.retrievablePost(uri, content, response, retry+1)
-			if err != nil {
-				return resp, err
+func (a *Core) retrievablePost(uri string, content []byte, response interface{}) (*http.Response, error) {
+	// during tests, allow to support ~90% of bad nonce with a minimum of attempts.
+	bo := backoff.NewExponentialBackOff()
+	bo.InitialInterval = 200 * time.Millisecond
+	bo.MaxInterval = 5 * time.Second
+	bo.MaxElapsedTime = 20 * time.Second
```
Comment (wyattjoh) [inline acme/api/api.go]: Maybe worth exporting these duration's from the package? Then it can be controllable by clients?

File: acme/api/api.go [hunk 2]
```diff
@@ -64,34 +67,46 @@ func (a *Core) post(uri string, reqBody, response interface{}) (*http.Response,
...
+	var resp *http.Response
+	operation := func() error {
+		var err error
+		resp, err = a.signedPost(uri, content, response)
+		if err != nil {
+			switch err.(type) {
+			// Retry if the nonce was invalidated
+			case *acme.NonceError:
+				log.Infof("nonce error retry: %s", err)
+				return err
```
Comment (wyattjoh) [inline acme/api/api.go]: `ctx` is not cancelled from this exit point, is that intentional?

Comment (ldez) [reply acme/api/api.go]: yes because I want to retry: if I cancel here i will not have retries.

File: acme/api/api.go (hunk 1 above)
Comment (ldez) [reply acme/api/api.go]: It's internal behavior, clients don't need to change these values.
If someone really needs to change that, we can export some values later.

Comment (ldez) [reply acme/api/api.go]: FYI before, with the system of 5 attempts, client waited around 1-2 seconds

**Pull request merged**